pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -149,6 +150,33 @@ public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService e
       TraceContext.register(requestId);
     }
 
+    int numConsuming = 0;
+    long minIndexTime = Long.MAX_VALUE;
+    long minIngestionTime = Long.MAX_VALUE;
+    // gather stats for realtime consuming segments
+    for (SegmentDataManager segmentMgr : segmentDataManagers) {
```

Comment:
should we be doing this after pruning segments?

Reply:
The idea is to get the stats for queried segments, so will keep it the same.

Reply:
For the question on ReduceService below, we currently don't get the queryId passed through to the BrokerReduceService. Will address that in a followup.